### ISSUE TITLE
Adding large number of charges in the cargo of a fit crashes Pyfa

### ISSUE THREAD

**Issue by Cosmitz** [pyfa-org/Pyfa#958]
I added a charge, right clicked, changed quantity to something large. I expected Pyfa would just max out my cargohold with howmuch of it could be entered, but that was not what happened. It became unresponsive. I closed it down.

Upon opening, the fit was not open. I reopened it and then it became unresponsive again. After multiple minutes it loaded the fit, but became unresponsive after clicking Cargo. Either way, it shows me i tried to load 2.65M m3 worth of what presumably is Federation Navy Antimatter Charge M into a 900m3 Gnosis. 

Eventually it times out, everytime i push cargo, and i can play with the fit some more but i had to delete it and recreate it to get the full functionality back.

**Comment by blitzmann** [comment]
I found this the other day. You probably entered something like `999999999999999999999`. What happens is that it exceeds the maximum INT value (2147483647) and freaks out.

What I'm not understanding is if it saves it to the database or not. Could you send me your database file `%userprofile%\.pyfa\saveddata.db` to sable.blitzmann@gmail.com

**Comment by blitzmann** [comment]
Actually, I believe what happens is that you stick in a value that is way too high, and then it fails when trying to get the price of the data:

```
            for cargo in fit.cargo:
                for _ in range(cargo.amount):
                    typeIDs.append(cargo.itemID)
```

I have no idea why we are adding the same typeID over and over again to get the price of the item, should only be once. I'll have to dig deeper. But this basically causes a crash because we're adding way too much to a list and trying to process it. Again, the database file would help in solving exactly what's going on here. :)

**Comment by Ebag333** [comment]
either way we should add a sane limit. Maybe 10k or something.

**Comment by Cosmitz** [comment]
https://dl.dropboxusercontent.com/u/4037018/saveddata.db

It was i think Alpha Gnosis Kill Astrahus. :P 

And no, the number wasn't that huge, .. *googles*

"2.65 million / 0.0125 =
two hundred twelve million"

There we go.. 212.000.000 pieces of good old FedNavy Antimatter. :D

**Comment by Ebag333** [comment]
that's still a large number lol.

**Comment by blitzmann** [comment]
I'm not seeing that in the database linked, but I can reproduce a Memory error by add 212 million to a cargo amount on my end.

**Comment by Cosmitz** [comment]
Mentioned that i forgot i had deleted and recreated (without adding 212 million ammo) the fit, thus useless for testing. But if the client freezes when you add said 212 million number, then it's the same thing. 

I recommend just doing a precheck of how much can be fit if you drag it to cargo, and filling it up to the brim if someone just goes 999999999. Alternatively, it'd be nice to have the same functionality on a ctrl+drag/drop or something like that, but that's outside the scope of this.

**Comment by blitzmann** [comment]
I am not a fan of only allowing up to a full cargo capacity. This creates issues if CCP ever reduces cargo capacity in one of their balancing passes, so we basically let the user decide and then tell them when they are over via the cargo resource

But, I just committed 3 fixes thanks to this issue:
* No longer looping through lord knows how many items based on the amount entered. This was the main problem, as it produced a `MemoryError` for very large amounts
* Max amount for cargo is now capped at `sys.maxint` which is 2,147,483,647. This will prevent us from overflowing the sqlite int limit
* A bug that prevented the cargo amount  window from closing if no amount was entered has been fixed.
